core/src/main/java/hudson/model/UpdateSite.java

```diff
@@ -731,6 +731,10 @@ public boolean isCompatibleWithInstalledVersion() {
                 else if (current.isOlderThan(requiredVersion)) {
                     deps.add(depPlugin);
                 }
+                // JENKINS-34494 - or if the plugin is disabled, this will allow us to enable it
+                else if (!current.isEnabled()) {
```

Comment:
It would, this spot just adds it to the list of dependencies to act on, later will have `deploy` called, which checks if it's disabled.